cub/test/catch2_test_device_select_if0.cu

```diff
@@ -0,0 +1,245 @@
+
+  // Needs to be device accessible
+  thrust::device_vector<int> num_selected_out(1, 0);
+  int    *d_num_selected_out = thrust::raw_pointer_cast(num_selected_out.data());
+
+  select_if(d_input,
+            d_output,
+            d_num_selected_out,
+            num_items,
+            always_true_t{});
+
+  REQUIRE(num_selected_out[0] == 0);
+}
+
+CUB_TEST("DeviceSelect::If handles all matched", "[device][select_if]", basic_types)
+{
+  using type = typename c2h::get<0, TestType>;
+
+  const int num_items = GENERATE_COPY(take(10, random(1, 1000000)));
+  thrust::device_vector<type> in(num_items);
+  thrust::device_vector<type> out(num_items);
+  c2h::gen(CUB_SEED(3), in);
+
+  type*  d_input  = thrust::raw_pointer_cast(in.data());
+  type*  d_output = thrust::raw_pointer_cast(out.data());
+
+  // Needs to be device accessible
+  thrust::device_vector<int> num_selected_out(1, 0);
+  int    *d_first_num_selected_out = thrust::raw_pointer_cast(num_selected_out.data());
+
+  select_if(d_input,
+            d_output,
+            d_first_num_selected_out,
+            num_items,
+            always_true_t{});
+
+  REQUIRE(num_selected_out[0] == num_items);
+  REQUIRE(out == in);
+}
+
+CUB_TEST("DeviceSelect::If handles no matched", "[device][select_if]", basic_types)
+{
+  using type = typename c2h::get<0, TestType>;
+
+  const int num_items = GENERATE_COPY(take(10, random(1, 1000000)));
+  thrust::device_vector<type> in(num_items);
+  c2h::gen(CUB_SEED(3), in);
+
+  type*  d_input  = thrust::raw_pointer_cast(in.data());
+  type*  d_output = nullptr;
+
+  // Needs to be device accessible
+  thrust::device_vector<int> num_selected_out(1, 0);
+  int    *d_first_num_selected_out = thrust::raw_pointer_cast(num_selected_out.data());
+
+  select_if(d_input,
+            d_output,
+            d_first_num_selected_out,
+            num_items,
+            always_false_t{});
+
+  REQUIRE(num_selected_out[0] == 0);
+}
+
+CUB_TEST("DeviceSelect::If does not change input", "[device][select_if]", basic_types)
+{
+  using type = typename c2h::get<0, TestType>;
+
+  const int num_items = GENERATE_COPY(take(10, random(1, 1000000)));
+  thrust::device_vector<type> in(num_items);
+  thrust::device_vector<type> out(num_items);
+  c2h::gen(CUB_SEED(3), in);
+
+  type*  d_input  = thrust::raw_pointer_cast(in.data());
+  type*  d_output = thrust::raw_pointer_cast(out.data());
+
+  // just pick one of the input elements as boundary
+  less_than_t<type> le{in[num_items / 2]};
+
+  // Needs to be device accessible
+  thrust::device_vector<int> num_selected_out(1, 0);
+  int    *d_first_num_selected_out = thrust::raw_pointer_cast(num_selected_out.data());
+
+  // copy input first
+  thrust::device_vector<type> reference = in;
+
+  select_if(d_input,
+            d_output,
+            d_first_num_selected_out,
+            num_items,
+            le);
+
+  REQUIRE(reference == in);
+}
+
+CUB_TEST("DeviceSelect::If is stable", "[device][select_if]", all_types)
```

Comment:
This test only makes sense with `custom_type`. Let's remove the type list and write `using type = custom_type...`.

Reply:
@miscco @elstehle in an offline discussion, we concluded that we'd like to be on a safe side and test some fundamental types here as well. I'm revoking my comment. Maybe we could at least not test against all types here.